### PR TITLE
Implements ability to override endpoint ordering

### PR DESCRIPTION
**Pull request by plroebuck**
Implements `--sorted` cmdline option (like Dredd).

Sorts endpoint requests in a sensible way so that objects are not modified before they are created.

Order: CONNECT, OPTIONS, POST, GET, HEAD, PUT, PATCH, DELETE, TRACE.

Closes: #22

### CODE REVIEW

**Comment by plroebuck** [comment]
@laurentperez, @nexus49, @joostmeijles, @misfitdavidl

Please provide feedback.

File: lib/add-tests.coffee [hunk 1]
```diff
@@ -1,5 +1,5 @@
 async = require 'async'
-_ = require 'underscore'
+_ = require 'lodash'
```
Comment (galk-in) [inline lib/add-tests.coffee]: It's fine for me to replace underscore with lodash. We should make it in every file including package.json. This changes help be consistent.

File: lib/add-tests.coffee [hunk 2]
```diff
@@ -52,6 +52,49 @@ addTests = (raml, tests, hooks, parent, callback, testFactory) ->
     # In case of issue #8, resource does not define methods
     resource.methods ?= []
 
+    if sortFirst && resource.methods.length > 1
```
Comment (galk-in) [inline lib/add-tests.coffee]: I'm thinking in [SOLID](https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)) paradigm. For me, ordering methods is not responsibility of add-test file. I think, we can make parser.coffee, which will be a wrapper on the raml-parser and have this functionality. In this case, the code will be more readable. What do you think?

File: lib/add-tests.coffee (hunk 1 above)
Comment (plroebuck) [inline lib/add-tests.coffee]: Replacement everywhere is longterm goal. But it's not exactly apples to apples. The lodash replacements have nuances (e.g., one deep copies when the other shallow copies), so I only replaced here where I could verify outcome.

File: lib/add-tests.coffee (hunk 2 above)
Comment (plroebuck) [inline lib/add-tests.coffee]: Think large chunks of **abao** could stand redesign. But I'd prefer having the functionality over refactoring it first. Would rather do suggested refactoring as different issue.

**Comment by galk-in** [comment]
Added https://github.com/cybertk/abao/issues/188 and https://github.com/cybertk/abao/issues/189

**Comment by galk-in** [comment]
👍 from me.
@plroebuck will we wait somebody for review?

**Comment by plroebuck** [comment]
It can wait for a couple days, in case someone else wants to say something. I asked several for feedback. I'm not expressly happy with having added YA parameter to addTests(), so it will give some additional time in case I can get around to the refactoriing as well.

**Comment by galk-in** [comment]
One week for code review finished.

**Pull request merged**